### PR TITLE
Remove noisy log statement

### LINKED ISSUE #9017: The new collector is emitting logs unnecessarily
**Describe the bug**
Version v0.90.0 of the collector is emitting noisy logs. This was reported by a user in the slack channel https://cloud-native.slack.com/archives/C01N6P7KR6W/p1701308425502839

```
2023-11-29T17:36:32.236-0800	info	exporterhelper/retry_sender.go:120	Exporting finished.	{"kind": "exporter", "data_type": "logs", "name": "splunk_hec"}
2023-11-29T17:36:44.053-0800	info	exporterhelper/retry_sender.go:120	Exporting finished.	{"kind": "exporter", "data_type": "logs", "name": "splunk_hec"}
2023-11-29T17:36:44.667-0800	info	exporterhelper/retry_sender.go:120	Exporting finished.	{"kind": "exporter", "data_type": "logs", "name": "splunk_hec"}
```

**Steps to reproduce**
Export some logs with the default logging level

**What did you expect to see?**
No noisy logs

**What did you see instead?**
The logs pasted above

**What version did you use?**
Version: v0.90.0

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Removes a noisy log statement

**Link to tracking Issue:**
Fixes #9017

### CODE REVIEW

File: exporter/exporterhelper/retry_sender.go [hunk 1]
```diff
@@ -117,7 +117,6 @@ func (rs *retrySender) send(ctx context.Context, req Request) error {
 			trace.WithAttributes(rs.traceAttribute, attribute.Int64("retry_num", retryNum)))
 
 		err := rs.nextSender.send(ctx, req)
-		rs.logger.Info("Exporting finished.", zap.Error(err))
```
Comment (TylerHelmuth) [inline exporter/exporterhelper/retry_sender.go:120]: Should we switch it to a Debug? Or would it be too noisy even for that?

Comment (codeboten) [reply exporter/exporterhelper/retry_sender.go:120]: Debug is probably ok

Comment (atoulme) [reply exporter/exporterhelper/retry_sender.go:120]: imo it's not useful. The error is logged if present a few lines below, and I don't see value in outputting a debug statement that an export is done, when we also have tracing and span events collected around those lines.

**Review by jpkrohling — APPROVED** [review]

**Comment by mx-psi** [comment]
```
Generated code is out of date, please run "make genotelcorecol" and commit the changes in this PR.
```

If we are going to do a bugfix release for this, we should follow the release procedure on https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/release.md#bugfix-release-procedure, which makes it less likely that we will get things wrong re:go dependencies

**Review by crobert-1 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@atoulme thanks for the fix!

**Pull request merged**